sdk/quantum/Azure.Quantum.Jobs/tests/QuantumJobClientTestEnvironment.cs

```diff
@@ -1,12 +1,126 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
 using Azure.Core.TestFramework;
+using Azure.Identity;
+using Newtonsoft.Json;
 
 namespace Azure.Quantum.Jobs.Tests
 {
     public class QuantumJobClientTestEnvironment : TestEnvironment
     {
+        private bool _initialized = false;
+        private readonly Lazy<AzLoginAccessTokenInfo> _azLoginAccessTokenInfo = new Lazy<AzLoginAccessTokenInfo>(QuantumJobClientTestEnvironment.GetAzLoginAccessTokenInfo);
+
         public string WorkspaceName => GetRecordedVariable("WORKSPACE_NAME");
+
+        public string GetRandomId(string idName)
+        {
+            var randomId = Guid.NewGuid().ToString("N");
+            var randomIdVariableName = $"RANDOM_ID_{idName}";
+            if (Mode == RecordedTestMode.Record)
+            {
+                Environment.SetEnvironmentVariable(randomIdVariableName, randomId);
+            }
+            return GetRecordedVariable(randomIdVariableName);
+        }
+
+        public new TokenCredential Credential
+        {
+            get
+            {
+                if (Mode == RecordedTestMode.Record)
+                {
+                    return new AzureCliCredential();
+                }
+                return base.Credential;
+            }
+        }
+
+        public QuantumJobClientTestEnvironment()
+        {
+        }
+
+        public void Initialize()
+        {
+            if (!_initialized
+                && Mode == RecordedTestMode.Record
+                && (GetOrSetVariable("LOCAL_RECORDING") != null))
```

Comment:
What's `LOCAL_RECORDING` for?  I think things like this will cause problems when someone outside your team needs to update your test recordings because of new Azure.Core features, etc.  We don't have to fix this before Preview, but it should be addressed soon.